cmd/cluster-agent/api/v1/kubernetes_metadata.go

```diff
@@ -16,8 +21,11 @@ func installKubernetesMetadataEndpoints(r *mux.Router) {
 	r.HandleFunc("/tags/pod/{nodeName}", getPodMetadataForNode).Methods("GET")
 	r.HandleFunc("/tags/pod", getAllMetadata).Methods("GET")
 	r.HandleFunc("/tags/node/{nodeName}", getNodeMetadata).Methods("GET")
+	r.HandleFunc("/cluster/id", getClusterID).Methods("GET")
 }
 
+func installCloudFoundryMetadataEndpoints(r *mux.Router) {}
```

Comment:
How was this working before if missing? 🤔

Reply:
it was working because this file didn't require the kubeapiserver build tag, so in the cloudfoundrymetadata code it had the function defined in this file.

What changed is that I added in this file the dependency on `pkg/util/kubernetes/apiserver/common` (L14) which requires `+build kubeapiserver` (needed to get the cluster ID). I didn't need it before because it was purely internal but now that we expose an endpoint with it, it gets surfaced in the api handlers here, and we need to add the build tag otherwise the CI is unhappy.

So I had to exclude the build tag in cloudfoundry, and now cloudfoundry needs an empty `installKubernetesMetadataEndpoints` and kubernetes needs an empty `installCloudFoundryMetadataEndpoints`. I've seen that in other parts of the code so i adopted the same technique.